pinot-spi/src/main/java/org/apache/pinot/spi/annotations/ScalarFunction.java

```diff
@@ -41,7 +41,7 @@
  *     - byte[]
  */
 @Retention(RetentionPolicy.RUNTIME)
-@Target(ElementType.METHOD)
+@Target({ElementType.METHOD, ElementType.TYPE})
```

Comment:
I would suggest to use AutoService instead of this annotation, but in case we still use it, I would remove the ElementType.METHOD, because we are only going to look for classes annotated with this, right?